include/zephyr/bluetooth/audio/audio.h

```diff
@@ -1521,10 +1529,11 @@ struct bt_audio_unicast_server_cb {
 	 *  @param stream  Stream object being reconfigured.
 	 *  @param qos     Quality of Service configuration.
 	 *
-	 *  @return 0 in case of success or negative value in case of error.
+	 *  @return structure containing response code and reason for
+	 *  the ASE Control operation.
 	 */
-	int (*qos)(struct bt_audio_stream *stream,
-		   const struct bt_codec_qos *qos);
+	struct bt_ascs_rsp (*qos)(struct bt_audio_stream *stream,
+				  const struct bt_codec_qos *qos);
```

Comment:
Hmm, I am a little on the fence w.r.t changing the return values of just these functions. Ideally the callbacks would be consistent in terms of return value. 

Wouldn't we want the same flexibility in the (re)config callback(s), as well as the `enable`? At least those seems to related to the `reason` value in the ASCS spec. 

Have you considered keeping `int` as the return value, and then use the `bt_ascs_rsp` as an additional parameter that the upper layers can fill in? 

Just FYI: I'm not saying that this isn't a good approach, I am just wondering if there are other approaches that may work better. @szymon-czapracki @MariuszSkamra any input?